lib/x86/matchfinder_impl.h

```diff
@@ -25,13 +25,27 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#ifdef __AVX2__
+#include "cpu_features.h"
+
+#undef DISPATCH_MATCHFINDER_AVX2
+#if (!defined(DEFAULT_MATCHFINDER_INIT) || !defined(DEFAULT_MATCHFINDER_REBASE)) && \
+    (defined(__AVX2__) || (X86_CPU_FEATURES_ENABLED &&	\
+		COMPILER_SUPPORTS_AVX2_TARGET_INTRINSICS))
 #  if MATCHFINDER_ALIGNMENT < 32
 #    undef MATCHFINDER_ALIGNMENT
 #    define MATCHFINDER_ALIGNMENT 32
 #  endif
+#  ifdef __AVX2__
+#    define ATTRIBUTES
+#    define DEFAULT_MATCHFINDER_INIT	matchfinder_init_avx2
+#    define DEFAULT_MATCHFINDER_REBASE	matchfinder_rebase_avx2
+#  else
+#    define ATTRIBUTES					__attribute__((target("avx2")))
```

Comment:
Note: this isn't compiling on the old Linux distribution test. I will fix this later.